ecommerce/api.py

```diff
@@ -458,12 +458,19 @@ def enroll_user_in_order_items(order):
         )
         edx_request_success = False
 
+    voucher = order.purchaser.vouchers.order_by("uploaded").last()
+    voucher_target = None
+    if voucher and not voucher.is_redeemed() and voucher.product is not None:
```

Comment:
Should this be:
```python
if voucher and voucher.is_redeemed() and voucher.product is not None:
```
I'm not getting `voucher.enrollment` assigned to anything unless I make that change.

Reply:
Oh...hmm....I suppose redeeming the voucher comes first. But that means we'll also have to make sure there isn't an enrollment already attached